rest/test_results_routes.go

```diff
@@ -14,8 +26,57 @@ package rest
 //
 // GET /testresults/test_name/{task_id}/{test_name}
 
-// Factory
+type testResultGetByTestNameHandler struct {
+	opts data.TestResultsOptions
+	sc   data.Connector
+}
 
-// Parse
+func makeGetTestResultByTestName(sc data.Connector) gimlet.RouteHandler {
+	return &testResultGetByTestNameHandler{
+		sc: sc,
+	}
+}
 
-// Run
+// Factory returns a pointer to a new testResultGetByTestNameHandler
+func (h *testResultGetByTestNameHandler) Factory() gimlet.RouteHandler {
+	return &testResultGetByTestNameHandler{
+		sc: h.sc,
+	}
+}
+
+// Parse fetches the task_id, test_name, and execution (if present)
+// from the http request.
+func (h *testResultGetByTestNameHandler) Parse(_ context.Context, r *http.Request) error {
+	catcher := grip.NewBasicCatcher()
+	var err error
+
+	h.opts.TaskID = gimlet.GetVars(r)["task_id"]
+	h.opts.TestName = gimlet.GetVars(r)["test_name"]
+	vals := r.URL.Query()
+	if len(vals[execution]) > 0 {
+		h.opts.Execution, err = strconv.Atoi(vals[execution][0])
+		h.opts.EmptyExecution = false
+		catcher.Add(err)
+	} else {
+		h.opts.EmptyExecution = true
+	}
+
+	return catcher.Resolve()
+}
+
+// Run calls FindFindTestResultByTestName and returns the test result.
```

Comment:
Typically, you don't need to say implementation details (i.e. you don't need to say it "calls FindTestResultByTestName") since people can read the code for that. You should just say the high level purpose of the function.